src/server/response.cpp

```diff
@@ -322,14 +314,16 @@ void Response::set_entry(const Entry& entry, const RequestContext& request) {
   set_mimeType(mimeType);
   set_cacheable();
 
+  // XXX: Which of the Accept-Encoding and Range headers has precedence if both
+  // XXX: are present? Can partial content be compressed?
```

Comment:
Searching (a bit) the internet, I've found nothing about this.
I would say that yes, partial content can be compressed.

I think we should resolve the byteRange first and then compress the content if needed.
However, our use case here is that we need byteRange for media (video) and they are not compressible.
I don't know if we should handle both case in the same time (but we must be sure to answer with the correct headers anyway)

Reply:
Hum... I'm not sure what I've reviewed. I've just comment this lines and github say me this is outdated...

Reply:
> Searching (a bit) the internet, I've found nothing about this.

Same for me. I guess I better have mentioned that in the PR description.

> I think we should resolve the byteRange first and then compress the content if needed.

The interaction of the `Accept-Encoding` and `Range` is a little tricky (and it's strange that there is no formal specification of what the correct interpretation should be). Suppose that the client issues a regular (non-range) request for a large compressible resource (e.g. a huge JSON or XML document). The resource is compressed and streamed to the client but the connection is interrupted at 95%. The client issues a range request to fetch the remaining 5% of the *compressed representation of the resource*, with the purpose of recombining it with whatever was sent in the first session and being able to decompress it. In such an scenario (which is different from your point of view of how `Range` should play with `Accept-Encoding`) the load on the server can be excessive - entire resource must be compressed with the purpose of sending only a small part of it.

Note that for partial/interrupted download of a compressed resource the client has no chance to decompress the received part, find out the byte-range for the uncompressed version of the resource and issue a range request for the remaining part.

That's why the safest bet is to *not* combine compression with partial responses.

Reply:
> Note that for partial/interrupted download of a compressed resource the client has no chance to decompress the received part, find out the byte-range for the uncompressed version of the resource and issue a range request for the remaining part

Why ? That's what I was thinking that what it has to be done.

For me the range is about the uncompressed content. If a connection is lost at 95%, client must uncompress the data it received and ask for what is left to have (in uncompressed range). This range will be compressed again in a new stream. But it cannot ask for the last 5% of the compressed stream.

Reply:
My assumption was that in general it is impossible to decompress a partial archive. That assumption is probably wrong for the compression algorithms/formats used in HTTP. Still, I doubt that if we implement that in kiwix-serve the handling on the client side will be like we expect. Therefore I think that we should not spend any time on this until we know of authoritative formal specification of how it must work.